### PR TITLE
Enable gesvda for ROCM >= 6.1

### PR DESCRIPTION
**Pull request by xinyazhang**
This also fixes a problem in `gesvd` driver when UV is not needed.

Fixes SWDEV-392430

### CODE REVIEW

**Comment by xinyazhang** [comment]
`PYTORCH_TESTING_DEVICE_ONLY_FOR="cuda" PYTORCH_TEST_WITH_ROCM=1 python test/test_linalg.py --use-pytest --color=yes` passed on MI210X locally.
With extra debugging output, the calling to gesvda backend function `apply_svd_cusolver_gesvdaStridedBatched`  is confirmed.

File: aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp [hunk 1]
```diff
@@ -332,7 +332,7 @@ inline static void svd_cusolver_gesvd(const Tensor& A, const Tensor& U, const Te
   // Similar to the case in svd_magma(), experiments have shown Vh tensor is
   // not guaranteed to be column major on ROCM, we have to create a copy to
   // deal with this
-  if (!not_A_H) {
+  if (compute_uv && !not_A_H) {
```
Comment (pruthvistony) [inline aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp:335]: Why is this change required?

Comment (pruthvistony) [reply aten/src/ATen/native/cuda/linalg/BatchLinearAlgebraLib.cpp:335]: I saw the comment now, when UV is not needed.

**Comment by pruthvistony** [comment]
Needs to be cherry-picked into upstream after 6.1 release and CI upgraded.

**Review by pruthvistony — APPROVED** [review]
